### ISSUE TITLE
Navigation Submenu block does not render menu item description

### ISSUE THREAD

**Issue by webmandesign** [WordPress/gutenberg#52505]
### Description

Unlike Navigation Link block, the Navigation Submenu block does not render menu item description HTML.

### Step-by-step reproduction instructions

1. In site editor create a navigational menu.
2. Add a submenu item(s) for one of the menu top level items.
3. Set a "Description" field for (at least) one of the submenu items.
4. Edit the parent (top level) menu item and set a "Description" field.
5. Save the edits and check the front-end display → the description text is not rendered for the parent menu item. The HTML rendering description text is missing.

### Environment info

- WP 6.2.2
- with and without Gutenberg 16.1.2

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by jordesign** [comment]
I can confirm that I see this happening (even through to 6.3beta 4).
I also noticed that it seems to affect _any_ parent menu item. I found that the same thing happened on the second level menu item if _it_ also had children.

**Comment by carolinan** [comment]
Hi
It is only visible if the theme supports it. It has been a while, but if I remember correctly this was a compromise that was made because classic themes do not show this description by default.

The theme needs to override the following CSS:

```
.wp-block-navigation .wp-block-navigation-item__description {
	display: none;
}
```

It looks like the submenu inner block is missing the condition for the description:
https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/navigation-submenu/index.php
compared to:
https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/navigation-link/index.php#L233

**Comment by carolinan** [comment]
I also think I opened an issue about making it easier to use the description, but I can't find it now. I will try to add it here if I can find it.

**Comment by webmandesign** [comment]
@carolinan Yes, you are correct, the theme has to opt in to display the menu item description. My theme does so. The issue is that the **HTML code for the description is not being rendered/inserted/produced**. No CSS will display the description if HTML code for it is missing :)

**Comment by getdave** [comment]
Missing this block of code. 

https://github.com/WordPress/gutenberg/blob/944e6b8c04a017c50af3f8862c9120bd7c383d4e/packages/block-library/src/navigation-link/index.php#L254-L259

**Comment by coderGtm** [comment]
Hi there...is this issue stale or does it need any discussions?

**Comment by webmandesign** [comment]
Hi @coderGtm, the issue is resolved in WordPress now. Thanks for reminding me about this ticket, I'm closing it now.
